CollAction/Controllers/ProjectsController.cs

```diff
@@ -396,16 +422,39 @@ public ViewResult StartInfo()
             return View(commitProjectViewModel);
         }
 
-        [HttpPost]
-        [ValidateAntiForgeryToken]
         [Authorize]
-        public async Task<IActionResult> Commit(int id, CommitProjectViewModel commitProjectViewModel)
+        public async Task<IActionResult> Commit(int? id)
         {
-            if (id != commitProjectViewModel.ProjectId)
+            if (id == null)
             {
                 return NotFound();
             }
 
+            var project =  await _service.GetProjectById(id); 
+            if (project == null)
+            {
+                return NotFound();
+            }
+            string validUrlForProjectName = WebUtility.UrlEncode(project.Name);
+            if(String.IsNullOrEmpty(validUrlForProjectName)){
+                return NotFound();
+            }
+            return LocalRedirect("~/projects/"+validUrlForProjectName+"/participate");
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        [Authorize]
+        public async Task<IActionResult> Commit(int id, CommitProjectViewModel commitProjectViewModel)
```

Comment:
Again, why so many endpoints for one action? We should only have the endpoints we're actually using.

Reply:
Ah, given that this is a url for logged-in users only, there's no need for duplication.